File: app/src/main/java/za/co/riggaroo/datecountdown/ui/event/list/EventAdapter.java
package za.co.riggaroo.datecountdown.ui.event.list;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import java.util.List;

import za.co.riggaroo.datecountdown.R;
import za.co.riggaroo.datecountdown.entity.Event;

public class EventAdapter extends RecyclerView.Adapter<EventViewHolder> {
    private final Context context;
    private List<Event> items;
    private View.OnClickListener deleteClickListener;
    private View.OnClickListener viewClickListener;

    EventAdapter(List<Event> items, Context context, View.OnClickListener viewClickListener, View.OnClickListener deleteClickListener) {
        this.items = items;
        this.context = context;
        this.deleteClickListener = deleteClickListener;
        this.viewClickListener = viewClickListener;
    }

    @Override
    public EventViewHolder onCreateViewHolder(ViewGroup parent,
                                              int viewType) {
        View v = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.list_item_event, parent, false);
        return new EventViewHolder(v);
    }

    @Override
    public void onBindViewHolder(EventViewHolder holder, int position) {
        Event item = items.get(position);
        holder.getEventTextView().setText(item.getName());
        holder.getDescriptionTextView().setText(item.getDescription());
        holder.getCountdownTextView().setText(context.getString(R.string.days_until, item.getDaysBetween()));
        holder.itemView.setTag(item);
        holder.getDeleteButton().setTag(item);
        holder.getDeleteButton().setOnClickListener(deleteClickListener);
        holder.itemView.setOnClickListener(viewClickListener);
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    void setItems(List<Event> events) {
        this.items = events;
        notifyDataSetChanged();
    }
}

File: app/src/main/java/za/co/riggaroo/datecountdown/ui/event/list/EventAdapter.kt
package za.co.riggaroo.datecountdown.ui.event.list

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import za.co.riggaroo.datecountdown.R
import za.co.riggaroo.datecountdown.entity.Event

class EventAdapter internal constructor(items: List<Event>, private val context: Context, private val viewClickListener: View.OnClickListener, private val deleteClickListener: View.OnClickListener) : RecyclerView.Adapter<EventViewHolder>() {

    var items: List<Event> = items
        set(newItems) {
            field = newItems
            notifyDataSetChanged()
        }

    override fun onCreateViewHolder(parent: ViewGroup,
                                    viewType: Int): EventViewHolder {
        val v = LayoutInflater.from(parent.context)
                .inflate(R.layout.list_item_event, parent, false)
        return EventViewHolder(v)
    }

    override fun onBindViewHolder(holder: EventViewHolder, position: Int) {
        val item = items[position]
        holder.eventTextView.text = item.name
        holder.descriptionTextView.text = item.description
        holder.countdownTextView.text = context.getString(R.string.days_until, item.getDaysBetween())
        holder.itemView.tag = item
        holder.deleteButton.tag = item
        holder.deleteButton.setOnClickListener(deleteClickListener)
        holder.itemView.setOnClickListener(viewClickListener)
    }

    override fun getItemCount() = items.size

}